### PR TITLE
chore(release): use docker image to facilitate update-copyright step

### PR DESCRIPTION
**Pull request by AndyZhang0707**
### Summary

It was not easy to install and figure out all the dependencies for the update-copyright script. This PR aims to install all the required dependencies into a docker image and use it to execute the update-copyright script.

### Checklist

- [ ] The Pull Request has tests
- [ ] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/blob/main/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

KAG-5297

Fix #_[issue number]_

### CODE REVIEW

File: scripts/release-lib.sh [hunk 1]
```diff
@@ -120,7 +120,10 @@ function commit_changelog() {
 function update_copyright() {
   version=$1
 
-  if ! "$scripts_folder/update-copyright"
+  DIR=$(dirname "$0")
+  PDIR=$(dirname "$DIR")
+
+  if ! (docker build -t kong/update-copyright ${DIR} && docker run -v ${PDIR}:/workspace --rm kong/update-copyright)
```
Comment (AndyZhang0707) [inline scripts/release-lib.sh:126]: We may consider publishing the docker image to reduce the build time in the future. But for now I've tried it several times docker always uses the local cached image thus it's not a big deal. @kikito any preferences?

File: scripts/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    && cd expat-${expat_version} \
+    && apt install -y build-essential \
+    && ./configure --prefix=/expat_lib \
+    && make && make install
+
+FROM ubuntu:latest
+
+COPY --from=expat-build /expat_lib /expat_lib
+
+RUN apt update && apt install -y curl libssl-dev libyaml-dev lua5.4 luarocks
```
Comment (Oyami-Srk) [inline scripts/Dockerfile:22]: The reason of not merging the `apt install` into previous stage is to keep dependencies up-to-date when we change the expat version, isn't it?

Comment (AndyZhang0707) [reply scripts/Dockerfile:22]: The tools installed in the expat-build stage are not in the final image. The reason I split it into two stages is that the expat-build stage has installed a lot of tools which we don't need in the final image.

**Review by kikito — APPROVED** [review]

**Review by Oyami-Srk — APPROVED** [review]

**Comment by kikito** [comment]
We don't need cherrypicking this into kong-ee, there's a different process there for copyright handling

**Pull request merged**